src/v/kafka/protocol/fetch.h

```diff
@@ -513,15 +513,26 @@ struct read_result {
     std::vector<cluster::rm_stm::tx_range> aborted_transactions;
 };
 
-using ntp_fetch_config = std::pair<model::materialized_ntp, fetch_config>;
+struct ntp_fetch_config {
+    ntp_fetch_config(
+      model::ntp ntp,
+      fetch_config cfg,
+      std::optional<model::ntp> materialized_ntp = std::nullopt)
+      : ntp(std::move(ntp))
+      , cfg(cfg)
+      , materialized_ntp(std::move(materialized_ntp)) {}
+    model::ntp ntp;
```

Comment:
It might be better to make an instance method `ntp()` that just returns `materialized_ntp.input_ntp()`, That way you save some space.

Reply:
`materialized_ntp` is not movable, I've introduced a little bit of redundancy in here but I need this type to be moveable.